### PR TITLE
Json file for de interleave function

### PR DESCRIPTION
**Pull request by ccominotti**

### CODE REVIEW

File: cgk_calcium_tools/files_io.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def write_log_file(
     input_files_keys : list, optional
         list with the parameters keys, by default ["input_movie_files"]
     output_file_key :  str, optional
-        key to access output file, by default "output_movie_files"
+        key to access output file, by default "output_movie_files". Relative path
```
Comment (ferchaure) [inline cgk_calcium_tools/files_io.py]: "Relative path" unclear

File: cgk_calcium_tools/filehandler.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from .processing import fix_frames
 import pandas as pd
 import shutil
-from .files_io import (
+from files_io import (
```
Comment (ferchaure) [inline cgk_calcium_tools/filehandler.py]: missing relative import

File: cgk_calcium_tools/filehandler.py [hunk 3]
```diff
@@ -289,33 +290,50 @@ def de_interleave(self, overwrite: bool = False) -> None:
...
                             os.remove(sp_file)
+                            if os.path.exists(json_file):
+                                os.remove(json_file)
                         else:
-                            existing_files.append(sp_file)
+                            if same_json_or_remove(
+                                parameters={
+                                    "main_file": main_file,
+                                    "planes_fs": planes_fs,
+                                    "focus": focus,
```
Comment (ferchaure) [inline cgk_calcium_tools/filehandler.py]: following the other functions, it's better to try to use same keys as function arguments. In this case: input_movie_files, output_movie_files, in_efocus_values

File: cgk_calcium_tools/filehandler.py [hunk 4]
```diff
@@ -289,33 +290,50 @@ def de_interleave(self, overwrite: bool = False) -> None:
...
-
                     except Exception as err:
                         print("Reading: ", main_file)
                         print("Writting: ", planes_fs)
                         raise err
 
+                    data = {
+                        "main_file": main_file,
+                        "planes_fs": planes_fs,
+                        "focus": focus,
```
Comment (ferchaure) [inline cgk_calcium_tools/filehandler.py]: same as previous comment

**Review by ferchaure — CHANGES_REQUESTED** [review]

File: cgk_calcium_tools/files_io.py (hunk 1 above)
Comment (ccominotti) [reply cgk_calcium_tools/files_io.py]: ok

**Review by ferchaure — APPROVED** [review]
great work, thanks

**Pull request merged**